tutorials/nservicebus-101/lesson-4/tutorial.md

```diff
@@ -0,0 +1,212 @@
+---
+title: "NServiceBus 101 Lesson 4: Publishing events"
+---
+
+So far in this course, we have only sent **commands** – one-way messages from a sender to a specific receiver. There's another type of message we have yet to cover called an **event**. In many ways events are just like commands. They're simple classes and you deal with them in much the same way. But from an architectural standpoint commands and events are *polar opposites*. This creates a useful dichotomy. We can take advantage of the properties of events to open up new possibilities in how we design software.
+
+In this lesson, we'll learn all about publishing events, and more importantly, what that allows us to do.
+
+
+## Objectives
+
+By the end of this lesson, you will have learned:
+
+ * The difference between commands and events
+ * How the Publish/Subscribe pattern helps to create more maintainable code
+ * How to define and name events
+ * How to publish an event
+ * How to subscribe to an event
+
+
+## What is an event?
+
+An **event** is another type of message that is published to multiple receivers, unlike a command which is sent to one receiver. Let's take a look at the formal definitions for commands and events:
+
+{{INFO:
+A **command** is a message that can be sent from one or more senders and is processed by a single receiver.
+
+An **event** is a message that is published from a single sender, and is processed by (potentially) many receivers.
+}}
+
+You can see that in many ways, commands and events are exact opposites, and the differences in their definition leads us to different uses for each.
+
+A command can be sent from anywhere, but is processed by one receiver. This is very similar to a web service, or any other [RPC](https://en.wikipedia.org/wiki/Remote_procedure_call)-style service. The big difference is that a one-way message does not have any return value like a web service would have. This means that the handler for the command is doing work for whomever is calling it, and that the sender has a very good idea about what it expects to happen as a result of sending the command. It is the sender saying "Will you please do something for me?" and so commands should be named in the [imperative tense](https://en.wikipedia.org/wiki/Imperative_mood), like `PlaceOrder`, `UpdateCustomerStatus`, and `ChargeCreditCard`. This creates tight coupling between the sender and receiver, because while it is possible to reject a command, you can't have true autonomy if someone else can tell you what to do.
```

Comment:
We're not trying to teach every possible option that the framework _can_ do, we're trying to teach the options that the student _needs to learn_. Because this lesson is an introduction to commands vs. events, sending a command to multiple endpoints is too advanced to be discussed.